opm/input/eclipse/Schedule/ResCoup/GrupSlav.hpp

```diff
@@ -0,0 +1,157 @@
+/*
+  Copyright 2024 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#ifndef RESERVOIR_COUPLING_GRUPSLAV_HPP
+#define RESERVOIR_COUPLING_GRUPSLAV_HPP
+
+#include <string>
+#include <map>
+#include "../HandlerContext.hpp"
```

Comment:
this is the cardinal sin of private headers. you cannot include a private header in a public header. and there is absolutely no need to include this header here either, just forward HandlerContext.

keep standard headers in alphabetical order.

Reply:
> just forward HandlerContext.

@akva2 What do you mean more explicitly by "forward" here? Obviously this file is using `HandlerContext` so wouldn't it be incomplete if I remove the include?

Reply:
```c++
class HandlerContext;
```
compiler doesn't need to know the definition of HandlerContext, only what type of beast it is. 

https://en.wikipedia.org/wiki/Forward_declaration